chatterbot/logic/mathematical_evaluation.py

```diff
@@ -77,9 +77,7 @@ def process(self, statement):
 
         # Use the result cached by the process method if it exists
         if input_text in self.cache:
-            cached_result = self.cache[input_text]
-            self.cache = {}
-            return cached_result
+            return self.cache[input_text]
```

Comment:
This change might cause a problem because it will keep caching statements forever until the program is restarted or the computer runs out of memory.